CoseSign1/DetachedSignatureFactory.cs

```diff
@@ -228,22 +377,57 @@ public DetachedSignatureFactory(HashAlgorithmName hashAlgorithmName, ICoseSign1M
             throw new ArgumentNullException("payload", "Either streamPayload or bytePayload must be specified, but not both at the same time, or both cannot be null");
         }
 
-        ReadOnlyMemory<byte> hash = streamPayload != null
-                                    ? InternalHashAlgorithm.ComputeHash(streamPayload)
-                                    : InternalHashAlgorithm.ComputeHash(bytePayload!.Value.ToArray());
+        ReadOnlyMemory<byte> hash;
+        string extendedContentType;
+        if (!payloadHashed)
+        {
+            hash = streamPayload != null
+                                 ? InternalHashAlgorithm.ComputeHash(streamPayload)
+                                 : InternalHashAlgorithm.ComputeHash(bytePayload!.Value.ToArray());

```

Comment:
should do a null check on bytePayload before using

Reply:
[Nullable ](https://devblogs.microsoft.com/dotnet/embracing-nullable-reference-types/) is supposed to eliminate the need for null checking within the library as it's described it's nullability requirements to the consumers.  It's a bit of a shift in way of thinking.

Since this is a private method, shifting the null checks into the public method would be the proper way to manage if we were going to add null checking.  That way the internals of the library never get exposed to the caller even under an exception.